src/source/view/controls/ControlsGroup.bs

```diff
@@ -149,6 +149,27 @@ namespace mv.controls
       return m.visibleChildren[m.selectedIndex]
     end function
 
+    function setFocusedControl(control as mc.types.node)
+      if type(control) = "roSGNode"
+        for i = 0 to m.visibleChildren.count() - 1
+          if m.visibleChildren[i].id = control.id
+            m.focusedIndex = i
```

Comment:
do we need to do anything beyond m.focuseDIndex = -i? I believe that does the m.redraw and updates m.focusedOcntorl - it has an onchange handler, right? please verify that's so. if so remove the next 2 lines of code. thanks.

Reply:
pushed changes. removed redraw call. I left the line to update the focusedControl, this was not updating when focusIndex was updated